### ISSUE TITLE
KeyError: 'index' on xclone.model.run_RDR function 

### ISSUE THREAD

**Issue by stublakemore** [single-cell-genetics/XClone#23]
Dear Xianjie,

Independent of my issue regarding eagle (xcltk baf - AssertionError #10), after successful completion of xcltk basefc pre-processing, and successful loading of the the RDR_adata AnnData object and associated .obs and .var parameters, upon running the xclone.model.run_RDR I receive the following error: "KeyError: 'index'". See attached for the complete execution of the code. Since it's in python, there aren't any folder substructures to send you log files, so hopefully the code is sufficient to help me out. 

Many thanks,

Stuart Blakemore

[xclone_RDR_KeyError.pdf](https://github.com/user-attachments/files/17746718/xclone_RDR_KeyError.pdf)

**Comment by stublakemore** [comment]
Dear Xianjie, 

Yes it's a conda environment with python 3.7 and XClone 0.3.5 as per the Github installation instructions (https://github.com/hxj5/XClone), although I used conda install rather than pip. Would you recommend that I rather install an earlier version of XClone? Maybe using conda rather than pip has caused the issue? 

Cheers,

Stuart

**Comment by hxj5** [comment]
Hi, thanks for the information. After quick google search, I found that the KeyError may arise from the incompatibility of the `anndata` packge, as discussed in [this thread](https://github.com/scverse/anndata/issues/726). 

While it is non-trivial to resolve the dependency issues (e.g., anndata), you may try uninstalling XClone and re-installing it via pip in current env. Alternatively, you can try installing the latest XClone version, e.g., XClone v0.3.8 under Python3.9, in a new conda env.

**Comment by stublakemore** [comment]
Hi, thanks for the quick response! I'll try a new conda env under Python3.9 installing XClone v0.3.8 and get back to you.

**Comment by stublakemore** [comment]
Hi Xianjie,

After fighting with this for a number of days, I finally found the solution! 

I installed Python3.9 and XClone v0.3.8 as you suggested, but based off of the solution to my xlctk baf pre-processing problem (here: https://github.com/hxj5/xcltk/issues/10#issuecomment-2472833470) I decided to try and use our high performance computer here instead of on my MacBook. 

In the end it didn't have anything to do with the anndata package, rather with out the index was loaded using the xclone.ppxclonedata function... Here's how the variables looked after loading:

>>> RDR_adata.var
      chr     start      stop      GeneName           GeneID arm chr_arm
0                                                                       
1       1     29554     31109    MIR1302-10  ENSG00000243485   p      1p
2       1     34554     36081       FAM138A  ENSG00000237613   p      1p
3       1     69091     70008         OR4F5  ENSG00000186092   p      1p
4       1     89295    133566  RP11-34P13.7  ENSG00000238009   p      1p
5       1     89551     91105  RP11-34P13.8  ENSG00000239945   p      1p
...    ..       ...       ...           ...              ...  ..     ...
32692   Y  27524447  27540866  LINC00266-4P  ENSG00000228786   q      Yq
32693   Y  27629055  27632852      CSPG4P1Y  ENSG00000240450   q      Yq
32694   Y  27768264  27771049          CDY1  ENSG00000172288   q      Yq
32695   Y  27874637  27879535         TTTY3  ENSG00000231141   q      Yq
32696   Y  28111776  28114889    AC007965.1  ENSG00000269393   q      Yq

This lead to the KeyError: 'index', because the index was misaligned with the RDR_adata.var columns. Running this code prior to solved the error:

 RDR_adata.var[:1] (because it slices based upon index labels rather than positional index), which lead to the following change in the RDR_adata.var structure

>>> RDR_adata.var
      chr     start      stop      GeneName           GeneID arm chr_arm
0       1     29554     31109    MIR1302-10  ENSG00000243485   p      1p
1       1     34554     36081       FAM138A  ENSG00000237613   p      1p
2       1     69091     70008         OR4F5  ENSG00000186092   p      1p
3       1     89295    133566  RP11-34P13.7  ENSG00000238009   p      1p
4       1     89551     91105  RP11-34P13.8  ENSG00000239945   p      1p
...    ..       ...       ...           ...              ...  ..     ...
32691   Y  27524447  27540866  LINC00266-4P  ENSG00000228786   q      Yq
32692   Y  27629055  27632852      CSPG4P1Y  ENSG00000240450   q      Yq
32693   Y  27768264  27771049          CDY1  ENSG00000172288   q      Yq
32694   Y  27874637  27879535         TTTY3  ENSG00000231141   q      Yq
32695   Y  28111776  28114889    AC007965.1  ENSG00000269393   q      Yq

Which then lead to the RDR_Xdata = xclone.model.run_RDR(RDR_adata, config_file = xconfig) code to run as expected!
